backend/code_groups/cg_controllers.js

```diff
@@ -93,9 +95,188 @@ exports.createGroup = async (req, res) => {
+  const { groupId } = req.params;
+  const userId = req.user.id;
+  try {
+    const isMember = await prisma.groupMember.findFirst({
+      where: {
+        groupId: parseInt(groupId),
+        userId: userId,
+      },
+    });
+    const isCreator = await prisma.codeGroup.findFirst({
+      where: {
+        id: parseInt(groupId),
+        creatorId: userId,
+      },
+    });
+
+    if (!isMember && !isCreator) {
+      return res.status(403).json({
+        error: "You do not have access to this group",
+      });
+    }
+    const group = await prisma.codeGroup.findUnique({
+      where: {
+        id: parseInt(groupId),
+      },
+      include: {
+        creator: true,
+        members: {
+          include: {
+            user: true,
+            addedBy: true,
+          },
+        },
+        files: true,
+      },
+    });
+    if (!group) {
+      return res.status(404).json({
+        error: "group not found",
+      });
+    }
+    res.status(200).json(group);
+  } catch (error) {
+    res.status(500).json({ message: "Error getting group details" });
+  }
+};
+
+//Get Random Image for Code Group
+exports.getGroupImageUrl = async (req, res) => {
+  const accessKey = process.env.IMAGE_API_ACCESS_KEY;
+  const { query, width, height } = req.query;
+
+  try {
+    const response = await axios.get("https://api.unsplash.com/search/photos", {
+      params: {
+        query: query,
+        per_page: 1,
+      },
+      headers: {
+        Authorization: `Client-ID ${accessKey}`,
+      },
+    });
+    if (response.data.results.length > 0) {
+      const image = await response.data.results[0];
+      const imageUrl = `${image.urls.raw}&w=${width}&h=${height}`;
+      res.status(200).json(imageUrl);
+    } else {
+      res.status(404).json({ message: "No result found" });
+    }
+  } catch (error) {
+    res.status(500).json({ message: "Error getting image" });
+  }
+};
+
+//Update Group Details: Group Name, Image and Add More Members
+exports.updateGroupDetails = async (req, res) => {
+  const { groupId } = req.params;
+  const { newGroupName, newGroupImageUrl, newMembers } = req.body;
+  const userId = req.user.id;
+  try {
+    //find group
+    const group = await prisma.codeGroup.findUnique({
+      where: {
+        id: parseInt(groupId),
+      },
+      include: {
+        creator: true,
+      },
+    });
+    if (!group) {
+      return res.status(404).json({
+        error: "Group not found",
+      });
+    }
+    if (group.creatorId !== userId) {
+      return res.status(403).json({
+        error: "You do not have permission to update this group",
+      });
+    }
+
+    let updatedGroupData = {};
+
+    if (newGroupName) {
+      updatedGroupData.groupName = newGroupName;
+    }
+    if (newGroupImageUrl) {
+      updatedGroupData.imgUrl = newGroupImageUrl;
+    }
+    const updatedGroup = await prisma.codeGroup.update({
+      where: {
+        id: parseInt(groupId),
+      },
+      data: updatedGroupData,
```

Comment:
Did you test that if `updatedGroupData` is empty or missing some fields, it doesn't erase the existing group data?